### DISCUSSION TITLE
What are the stage two weights and early stopping

### DISCUSSION THREAD

**Discussion by nihil39** [ACEsuit/mace#895] [Q&A] [answered]
Hi, maybe this is a trivial question. I am running a model with stage one weight of 10, 1, 1, for Energy, forces and stress

`INFO: WeightedEnergyForcesStressLoss(energy_weight=10.000, forces_weight=1.000, stress_weight=1.000)
`
I am leaving the default parameters for stage two but I don't understand what are the weights in that phase of training. 

This is what I can read from the log file

`Stage Two (after 200 epochs) with loss function: WeightedEnergyForcesStressLoss(energy_weight=1000.000, forces_weight=100.000, stress_weight=10.000), energy weight : 1000.0, forces weight : 100.0, stress weight : 1.0 and learning rate : 0.001`

Are the 1000, 100 and 10 or 1000, 100 and 1 ?

Other question, what it is the parameter that controls early stopping? I mean the maximum number of epochs without a decreasing in the loss function? Is it just the patience value?

Thank you.

**Comment by vue1999** [comment, marked as answer]
Hi, thanks for noticing this inconsistency. The correct weights for the second stage are 1000, 100, and 10 for energy, forces, and stress respectively. There was a mistake in the logging, which [PR #292](https://github.com/ACEsuit/mace/pull/896) will fix.
Regarding your question about early stopping, the parameter that controls this is the "patience" value. This determines the maximum number of epochs to wait without improvement in the validation loss before stopping training or moving on to stage two.
